cloud/deployment/deployment.go

```diff
@@ -520,6 +519,13 @@ func Create(name, workspaceID, description, clusterID, runtimeVersion, dagDeploy
 	return nil
 }
 
+func getDefaultNodePoolID(nodePools *[]astroplatformcore.NodePool) string {
```

Comment:
why do we even use a pointer to an array on the Core side 🤯

Reply:
Ha, we don't, it's just generated like this on the CLI side because the field is optional in the API spec.